core/src/term/pattern/compile.rs

```diff
@@ -668,23 +668,28 @@ impl Compile for MatchData {
     //      # this primop evaluates body with an environment extended with bindings_id
     //      %pattern_branch% body bindings_id
     fn compile(mut self, value: RichTerm, pos: TermPos) -> RichTerm {
-        if self.branches.iter().all(|(pat, _)| {
```

Comment:
For the TagsOnlyMatch optimization, don't you also need to check that there are no guards?

Reply:
Ah, you're right (although we could still first do the hashmap access and then check the guard, but it's out of scope for this PR and also probably much less useful/common). Thanks